src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -868,7 +868,7 @@ public ICompilationTracker WithDoNotCreateCreationPolicy(CancellationToken cance
 
                 foreach (var result in driverRunResult.Results)
                 {
-                    if (!IsGeneratorRunResultToIgnore(result))
+                    if (!result.Diagnostics.IsDefaultOrEmpty && !IsGeneratorRunResultToIgnore(result))
```

Comment:
i feel lke this needs comments expalining things.  also, intention that this check is not in IsGeneratorRunResultToIgnore?